tests/nn/data_parallel/test_fsdp_regnet.py

```diff
@@ -143,39 +227,65 @@ def _test_func(
     if fsdp_config:
         ddp = False
         assert isinstance(fsdp_config, dict), str(fsdp_config)
+        if fsdp_config["mixed_precision"]:
+            # To match DDP in AMP -O1, we need fp32 reduce scatter.
+            fsdp_config["fp32_reduce_scatter"] = True
 
     model = Model()
     model.load_state_dict(state_before)
     model = model.cuda()
 
+    class DummyScaler:
```

Comment:
@anj-s I used this DummyScaler when scaling is not needed in full precision mode. Could be a useful thing to do in general to unify full and mixed precision training loop.